R/test-package.r

```diff
@@ -49,19 +49,27 @@ test_package <- function(package, filter = NULL, reporter = "summary") {
   test_path2 <- file.path(test_path, "testthat")
   if (file.exists(test_path2)) test_path <- test_path2
 
-  reporter <- find_reporter(reporter)
+  res <- .run_tests(package, test_path, filter, reporter)
+
+  invisible(res)
+}
 
+
+.run_tests <- function(package, test_path, filter, reporter)
+{
+  reporter <- find_reporter(reporter)
   env <- test_pkg_env(package)
-  with_top_env(env, {
-    df <- test_dir(test_path, reporter = reporter, env = env, filter = filter)
+  res <- with_top_env(env, {
+    test_dir(test_path, reporter = reporter, env = env, filter = filter)
   })
 
-  if (reporter$failed) {
+  if (!all_passed.testthat_results(res)) {
```

Comment:
Can you please either make `all_passed` a proper S3 generic, or just rename `all_passed.testthat_results` to `all_passed`